test/e2e/nsx_ipaddressallocation_test.go

```diff
@@ -0,0 +1,132 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+)
+
+const (
+	externalCIDR   = "192.168.0.0/27"
+	privateCIDR    = "172.26.0.0/27"
+	privateTGWCIDR = "10.246.0.0/27"
```

Comment:
I'm not sure if there will be other test case using IPBlock, if it is difficult to find expected cidr, we can just make sure the cidr is subnet of the whole IPBlock.

Reply:
ok